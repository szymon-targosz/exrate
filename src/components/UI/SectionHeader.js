import React from 'react';
import PropTypes from 'prop-types';

const SectionHeader = ({ children, cssClasses, title, ...rest }) => {
   let classes = [ 'section-header' ];

   if (cssClasses) classes = classes.concat(cssClasses);

   return (
      <header className={classes.join(' ')} {...rest}>
         <h2 className="heading-2">{title}</h2>
         {children}
      </header>
   );
};

SectionHeader.propTypes = {
   cssClasses: PropTypes.oneOfType([ PropTypes.string, PropTypes.array ]),
   title: PropTypes.string.isRequired
};

export default SectionHeader;
