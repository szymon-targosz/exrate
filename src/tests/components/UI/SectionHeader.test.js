import React from 'react';
import { shallow } from 'enzyme';
import SectionHeader from '../../../components/UI/SectionHeader';

let wrapper;

beforeEach(() => {
   wrapper = shallow(<SectionHeader title="Some interesting text" children="CHILDREN HERE" />);
});

describe('COMPONENT - SectionHeader', () => {
   test('Should render SectionHeader correctly', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('Should add external css classes - array', () => {
      wrapper.setProps({ cssClasses: [ 'bg-secondary', 'border', 'border-info' ] });
      expect(wrapper).toMatchSnapshot();
   });

   test('Should add external css classes - string', () => {
      wrapper.setProps({ cssClasses: 'bg-warning p-s' });
      expect(wrapper).toMatchSnapshot();
   });
});
