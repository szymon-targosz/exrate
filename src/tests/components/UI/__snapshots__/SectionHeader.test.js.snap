// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SectionHeader Should add external css classes - array 1`] = `
<header
  className="section-header bg-secondary border border-info"
>
  <h2
    className="heading-2"
  >
    Some interesting text
  </h2>
  CHILDREN HERE
</header>
`;

exports[`COMPONENT - SectionHeader Should add external css classes - string 1`] = `
<header
  className="section-header bg-warning p-s"
>
  <h2
    className="heading-2"
  >
    Some interesting text
  </h2>
  CHILDREN HERE
</header>
`;

exports[`COMPONENT - SectionHeader Should render SectionHeader correctly 1`] = `
<header
  className="section-header"
>
  <h2
    className="heading-2"
  >
    Some interesting text
  </h2>
  CHILDREN HERE
</header>
`;
